wgpu-native/src/lib.rs

```diff
@@ -168,46 +168,100 @@ pub struct ByteArray {
+            }
+        }
+    )
+}
+
+transparent!(pub struct InstanceId(hub::Id););
+new_id!(InstanceId);
 type InstanceHandle = back::Instance;
-pub type AdapterId = hub::Id;
+transparent!(pub struct AdapterId(hub::Id););
 type AdapterHandle = hal::Adapter<back::Backend>;
-pub type DeviceId = hub::Id;
+transparent!(pub struct DeviceId(hub::Id););
 type DeviceHandle = Device<back::Backend>;
-pub type QueueId = DeviceId;
-pub type BufferId = hub::Id;
+transparent!(pub struct QueueId(DeviceId););
+transparent!(pub struct BufferId(hub::Id););
+new_id!(BufferId);
+to_id!(BufferId);
 type BufferHandle = Buffer<back::Backend>;
 // Resource
-pub type TextureViewId = hub::Id;
+transparent!(pub struct TextureViewId(hub::Id););
+new_id!(TextureViewId);
 type TextureViewHandle = TextureView<back::Backend>;
-pub type TextureId = hub::Id;
+transparent!(pub struct TextureId(hub::Id););
+new_id!(TextureId);
+to_id!(TextureId);
 type TextureHandle = Texture<back::Backend>;
-pub type SamplerId = hub::Id;
+transparent!(pub struct SamplerId(hub::Id););
 type SamplerHandle = Sampler<back::Backend>;
 // Binding model
-pub type BindGroupLayoutId = hub::Id;
+transparent!(pub struct BindGroupLayoutId(hub::Id););
 type BindGroupLayoutHandle = BindGroupLayout<back::Backend>;
-pub type PipelineLayoutId = hub::Id;
+transparent!(pub struct PipelineLayoutId(hub::Id););
 type PipelineLayoutHandle = PipelineLayout<back::Backend>;
-pub type BindGroupId = hub::Id;
+transparent!(pub struct BindGroupId(hub::Id););
 type BindGroupHandle = BindGroup<back::Backend>;
 // Pipeline
-pub type InputStateId = hub::Id;
-pub type ShaderModuleId = hub::Id;
+transparent!(pub struct InputStateId(hub::Id););
+transparent!(pub struct ShaderModuleId(hub::Id););
 type ShaderModuleHandle = ShaderModule<back::Backend>;
-pub type RenderPipelineId = hub::Id;
+transparent!(pub struct RenderPipelineId(hub::Id););
 type RenderPipelineHandle = RenderPipeline<back::Backend>;
-pub type ComputePipelineId = hub::Id;
+transparent!(pub struct ComputePipelineId(hub::Id););
 type ComputePipelineHandle = ComputePipeline<back::Backend>;
 // Command
-pub type CommandBufferId = hub::Id;
+transparent!(pub struct CommandBufferId(hub::Id););
 type CommandBufferHandle = CommandBuffer<back::Backend>;
-pub type CommandEncoderId = CommandBufferId;
-pub type RenderPassId = hub::Id;
+transparent!(pub struct CommandEncoderId(CommandBufferId););
+impl CommandEncoderId {
+    pub fn to_command_buffer(&self) -> CommandBufferId {
+        self.0
+    }
+}
+transparent!(pub struct RenderPassId(hub::Id););
 type RenderPassHandle = RenderPass<back::Backend>;
-pub type ComputePassId = hub::Id;
+transparent!(pub struct ComputePassId(hub::Id););
 type ComputePassHandle = ComputePass<back::Backend>;
 // Swap chain
-pub type SurfaceId = hub::Id;
+transparent!(pub struct SurfaceId(hub::Id););
 type SurfaceHandle = Surface<back::Backend>;
-pub type SwapChainId = SurfaceId;
+transparent!(pub struct SwapChainId(SurfaceId););
```

Comment:
I think it should be `hub::Id` here, although it depends on whether we'll manage to have the strong typing integrated into `HUB` itself.